src/content-helper/dashboard-page/pages/traffic-boost/preview/preview.tsx

```diff
@@ -0,0 +1,406 @@
+				action: 'parsely_post_preview',
+				post_id: activePost.id,
+				_wpnonce: window._parsely_traffic_boost_preview_nonce ?? '',
+			} )
+			: addQueryArgs( activePost.guid.raw, {
+				parsely_preview: 'true',
+				_wpnonce: window._parsely_traffic_boost_preview_nonce ?? '',
+			} );
+
+		// Only set loading state if URL actually changes.
+		if ( newUrl !== previewUrl ) {
+			setIsLoading( true );
+			setPreviewUrl( newUrl );
+		}
+	}, [ activePost, isFrontendPreview, previewUrl ] );
+
+	/**
+	 * Opens the post in a new tab.
+	 *
+	 * @since 3.18.0
+	 */
+	const openPostInNewTab = () => {
+		if ( ! activePost?.guid?.raw ) {
+			return;
+		}
+
+		window.open( activePost.guid.raw, '_blank' );
+	};
+
+	/**
+	 * Opens the post editor in a new tab.
+	 *
+	 * @since 3.18.0
+	 */
+	const openPostEditor = () => {
+		if ( ! activePost?.id ) {
+			return;
+		}
+
+		window.open( `${ window.location.origin }/wp-admin/post.php?post=${ activePost.id }&action=edit`, '_blank' );
+	};
+
+	/**
+	 * Opens the Parse.ly dashboard for this post in a new tab.
+	 *
+	 * @since 3.18.0
+	 */
+	const openParselyDashboard = () => {
+		if ( ! activePost?.guid?.raw ) {
+			return;
+		}
+
+		const parselyDashboardUrl = `https://dash.parsely.com/${ window.wpParselySiteId }/find?url=${ encodeURIComponent( activePost.guid.raw ) }`;
+		window.open( parselyDashboardUrl, '_blank' );
+	};
+
+	/**
+	 * Handles the next item event.
+	 *
+	 * @since 3.18.0
+	 */
+	const handleNext = () => {
+		let nextItem: TrafficBoostLink | undefined;
+
+		if ( isInboundLink ) {
+			nextItem = inboundLinks?.[ itemIndex ];
+		} else {
+			nextItem = suggestions?.[ itemIndex ];
+		}
+		if ( nextItem ) {
+			setItemIndex( itemIndex + 1 );
+			setSelectedLink( nextItem );
+		}
+	};
+
+	/**
+	 * Handles the previous item event.
+	 *
+	 * @since 3.18.0
+	 */
+	const handlePrevious = () => {
+		let previousItem: TrafficBoostLink | undefined;
+
+		if ( isInboundLink ) {
+			previousItem = inboundLinks?.[ itemIndex - 2 ];
+		} else {
+			previousItem = suggestions?.[ itemIndex - 2 ];
+		}
+
+		if ( previousItem ) {
+			setItemIndex( itemIndex - 1 );
+			setSelectedLink( previousItem );
+		}
+	};
+
+	/**
+	 * Handles the accept event.
+	 *
+	 * @since 3.18.0
+	 *
+	 * @param {TrafficBoostLink} link The link to accept.
+	 */
+	const handleAccept = async ( link: TrafficBoostLink ) => {
+		setIsAccepting( link, true );
+
+		// Accept the suggestion.
+		const acceptedLink = await onAccept( link );
+
+		// Remove suggestion from the list.
+		removeSuggestion( link );
+
+		// Add the link to the inbound links list.
+		addInboundLink( acceptedLink );
+
+		setIsAccepting( link, false );
+
+		// Show a snackbar success message.
+		createSuccessNotice(
+			__( 'Link planted on', 'wp-parsely' ) + ' ' + activePost.title.rendered,
+			{
+				type: 'snackbar',
+				icon: <Icon icon={ linkIcon } />,
+			}
+		);
+
+		// If the accepted link is the last one, move to the inbound tab and select the accepted link.
```

Comment:
I think this comment might need some rephrasing, as it is too similar with the comment in the `else if`. Same for the additional occurrences of this.